firedrake/constant.py

```diff
@@ -197,3 +205,32 @@ def __idiv__(self, o):
 
     def __str__(self):
         return str(self.dat.data_ro)
+
+
+# Unicode handler for Firedrake constants
+def _unicode_format_firedrake_constant(self, o):
+    """Format a Firedrake constant."""
+    i = o.count()
+    var = "💩"
+    if len(o.ufl_shape) == 1:
+        var += UC.combining_right_arrow_above
+    elif len(o.ufl_shape) > 1 and self.colorama_bold:
+        var = f"{colorama.Style.BRIGHT}{var}{colorama.Style.RESET_ALL}"
+    return f"{var}{subscript_number(i)}"
+
+
+# This monkey patches ufl2unicode support for Firedrake constants
+Expression2UnicodeHandler.firedrake_constant = _unicode_format_firedrake_constant
```

Comment:
I don't think that's the right place to add `ufl2unicode` support for constants, see my comment [here](https://github.com/firedrakeproject/firedrake/issues/3224).

Reply:
I agree, this code is disgusting. I envision my code as being a temporary workaround current issues whilst someone with more UFL skills fixes this upstream.